lib/LedgerSMB/Scripts/setup.pm

```diff
@@ -775,27 +775,54 @@ verify_check => md5_hex($check->test_query),
     $sth->finish();
 
     my $heading = { map { $_ => $_ } @{$check->display_cols} };
-    my $buttons = [
+    my %buttons = map { $_ => 1 } @{$check->buttons};
+    my $buttons;
+    push @$buttons,
            { type => 'submit',
              name => 'action',
-            value => $check->columns ? 'fix_tests' : 'cancel',
-             text => $request->{_locale}->text($check->columns
-                                                ? 'Save and Retry'
-                                                : 'Cancel'),
+            value => 'fix_tests',
+          tooltip => { id => 'action-fix-tests',
+                       msg => $request->{_locale}->text($check->{tooltips}{'Save and Retry'}),
+                       position => 'above'
+                     },
+             text => $request->{_locale}->text('Save and Retry'),
             class => 'submit' },
```

Comment:
There's a stray comma (value separator) here (left over from when this was an array).